src/vm/checker/typecheck/interface.rs

```diff
@@ -0,0 +1,215 @@
+use std::collections::BTreeMap;
+
+use vm::types::{TypeSignature, FunctionArg, AtomTypeIdentifier, TupleTypeSignature};
+use vm::checker::typecheck::FunctionType;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum ContractInterfaceFunctionAccess {
+    private,
+    public,
+    read_only,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct ContractInterfaceTupleType {
+    pub name: String,
+    pub r#type: ContractInterfaceAtomType,
```

Comment:
Can you choose different variable names that don't require the `r#` prefix?

Reply:
Yep will do, these seemed odd. 
I'm also looking for feedback on other names like `access`.